### PR TITLE
fix: include `traces` field when running `forge test -vvvv --json`

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Closes: https://github.com/foundry-rs/foundry/issues/6922

Part of a larger meta-ticket: https://github.com/foundry-rs/foundry/issues/8794

## Solution

Enable and includes traces in JSON output when `forge test` is ran with `--json` enabled (and verbosity >= 3)

Includes Snapbox tests

Future improvements could include making `TraceWriter` in `revm-inspectors` JSON compatible so we return decoded traces and well as https://github.com/foundry-rs/foundry/issues/7813 once that is added.

This should not impact the JUnit XML reports as that follows a strict schema already

### CODE REVIEW

**Review by grandizzy — APPROVED** [review]
lgtm

File: crates/forge/tests/cli/test_cmd.rs [hunk 1]
```diff
@@ -274,6 +274,55 @@ Ran 1 test suite [ELAPSED]: 1 tests passed, 0 failed, 0 skipped (1 total tests)
...
+
+    // Disable optimizer because for simple contract most functions will get inlined.
+    prj.write_config(Config { optimizer: false, ..Default::default() });
+
+    prj.add_test("Simple.t.sol", SIMPLE_CONTRACT).unwrap();
+
+    // Assert that without verbose output the json output does not include the traces
+    cmd.args(["test", "--json"])
+        .assert_success()
+        .stdout_eq(file!["../fixtures/SimpleContractTestNonVerbose.json"]);
```
Comment (DaniPopes) [inline crates/forge/tests/cli/test_cmd.rs:323]:
```suggestion
        .stdout_eq(file!["../fixtures/SimpleContractTestNonVerbose.json"; Json]);
```

File: crates/forge/tests/cli/test_cmd.rs [hunk 2]
```diff
@@ -274,6 +274,55 @@ Ran 1 test suite [ELAPSED]: 1 tests passed, 0 failed, 0 skipped (1 total tests)
...
+
+    // Disable optimizer because for simple contract most functions will get inlined.
+    prj.write_config(Config { optimizer: false, ..Default::default() });
+
+    prj.add_test("Simple.t.sol", SIMPLE_CONTRACT).unwrap();
+
+    // Assert that with verbose output the json output includes the traces
+    cmd.args(["test", "-vvv", "--json"])
+        .assert_success()
+        .stdout_eq(file!["../fixtures/SimpleContractTestVerbose.json"]);
```
Comment (DaniPopes) [inline crates/forge/tests/cli/test_cmd.rs:308]:
```suggestion
        .stdout_eq(file!["../fixtures/SimpleContractTestVerbose.json"; Json]);
```

File: crates/forge/tests/cli/test_cmd.rs [hunk 3]
```diff
@@ -274,6 +274,55 @@ Ran 1 test suite [ELAPSED]: 1 tests passed, 0 failed, 0 skipped (1 total tests)
...
+    function test() public {
+        SimpleContract c = new SimpleContract();
+        c.setValues(100);
+    }
+}
+   "#;
+
+forgetest_init!(can_run_test_with_json_output_verbose, |prj, cmd| {
+    prj.wipe_contracts();
+    prj.clear();
```
Comment (DaniPopes) [inline crates/forge/tests/cli/test_cmd.rs:298]: use `forgetest!` without init if you're not going to use the initial project layout, we should go through existing tests to check for this too

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]
nice, pending nits

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**